sktime/performance_metrics/forecasting/probabilistic/_classes.py

```diff
@@ -0,0 +1,225 @@
+    values are better.
+    """
+
+    _tags = {"scitype:y_pred": "quantiles"}
+
+    def __init__(self, func, name=None, multioutput="uniform_average"):
+        self.multioutput = multioutput
+        super().__init__(func, name=name)
+
+    def __call__(self, y_true, y_pred, **kwargs):
+        """Calculate metric value using underlying metric function.
+
+        Parameters
+        ----------
+        y_true : pd.Series, pd.DataFrame or np.array of shape (fh,) or \
+                (fh, n_outputs) where fh is the forecasting horizon
+            Ground truth (correct) target values.
+
+        y_pred : pd.Series, pd.DataFrame or np.array of shape (fh,) or  \
+                (fh, n_outputs)  where fh is the forecasting horizon
+            Forecasted values.
+
+        Returns
+        -------
+        loss : float
+            Calculated loss metric.
+        """
+        return self.evaluate(y_true, y_pred, multioutput=self.multioutput, **kwargs)
+
+    def evaluate(self, y_true, y_pred, multioutput, **kwargs):
+        """Evaluate the desired metric on given inputs."""
+        # Input checks
+        _, y_true, y_pred, multioutput = _check_reg_targets(y_true, y_pred, multioutput)
+        # y_pred = _check_y_pred_type()
+
+        return self._evaluate(y_true, y_pred, multioutput, **kwargs)
+
+    def _evaluate(self, y_true, y_pred, multioutput, **kwargs):
+        raise NotImplementedError
+
+    def evaluate_by_index(self, y_true, y_pred, multioutput, **kwargs):
+        """Return the metric evaluated at each time point."""
+        _, y_true, y_pred, multioutput = _check_reg_targets(y_true, y_pred, multioutput)
+        # y_pred = _check_y_pred_type(self)
+
+        return self._evaluate_by_index(y_true, y_pred, multioutput)
+
+    def _evaluate_by_index(self, y_true, y_pred, multioutput, **kwargs):
+        n = y_true.shape[0]
+        out_series = pd.Series(index=y_pred.index)
+        x_bar = self.evaluate(y_true, y_pred, multioutput, **kwargs)
+        for i in range(n):
+            out_series[i] = n * x_bar - (n - 1) * self.evaluate(
+                np.vstack((y_true[:i, :], y_true[i + 1 :, :])),
+                np.vstack((y_pred[:i, :], y_pred[i + 1 :, :])),
+                multioutput,
+            )
+        return out_series
+
+
+class PinballLoss(_BaseProbForecastingErrorMetric):
+    """Evaluate the pinball loss at the alpha quantile.
+
+    Parameters
+    ----------
+    alpha : The significance level required for the pinball loss
+
+    multioutput : string "uniform_average" or "raw_values" determines how multioutput
+    results will be treated
+    """
+
+    def __init__(self, alphas, multioutput="uniform_average"):
+        name = "PinballLoss"
+        # _tags = {"scitype:pred": "quantile"}
+        func = pinball_loss
+        self.alphas = self._check_alpha(alphas)
+        super().__init__(func=func, name=name, multioutput=multioutput)
+
+    def _check_alpha(self, alpha):
```

Comment:
would it make sense to put this in a quantile mixin, or in the base class? Other quantile losses will need this too?